feat: ROOT-29: Build control tags dropdown for filtering labels within a specific tag

matt-bernstein opened the pull request:
This PR includes the FE changes for the LSE feature https://humansignal.atlassian.net/browse/ROOT-13.

Change to columns:
- introduce `annotations_results_json` columns that are only used for filtering, don't show up in DM and are not orderable or selectable
- finish implementation of `orderable` flag; did not fully work as intended

Change to filters:
- Limit available operators for new columns to ones that make sense given BE implementation
- Fix search in filter name selector to work on column name instead of column id, to remove false positive matches
- Clear user input on selecting a different filter, to fix bug where user input sent to BE leaked across filters

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

matt-bernstein committed:
feat: ROOT-29: Build control tags dropdown for filtering labels within a specific tag

matt-bernstein committed:
wip add control tag pseudocolumn

matt-bernstein committed:
Revert "wip add control tag pseudocolumn"

This reverts commit f04db2b7551d51e189fff048acc7504d2179ba8a.

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-ROOT-29

matt-bernstein committed:
use orderable and internal flags on column

yyassi-heartex commented on web/libs/datamanager/src/stores/Tabs/store.js:
```diff
@@ -456,6 +458,9 @@ export const TabStore = types
             field: columnID,
             schema: col.schema ?? null,
           });
+          // if (col.internal) {
+          //   return;
+          // }
```
```suggestion
```

matt-bernstein replied:
still a WIP, keeping this around until the final pass in case I need it later :)

yyassi-heartex commented on web/libs/datamanager/src/stores/Tabs/store.js:
```diff
@@ -91,8 +91,10 @@ export const TabStore = types
-
-      return cols.get(self.selected?.target ?? "tasks") ?? [];
+      const list = cols.get(self.selected?.target ?? "tasks") ?? [];
+      // // Filter out internal columns so they never reach grid/column selector
+      // return list.filter((c) => !c.internal);
```
```suggestion
```

matt-bernstein committed:
ff

matt-bernstein committed:
pipe through special string filter

matt-bernstein committed:
move ff

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-ROOT-29

matt-bernstein committed:
dead code

matt-bernstein committed:
limit filter operators

matt-bernstein committed:
update filter

matt-bernstein committed:
change placeholder

matt-bernstein committed:
simplify user input handling

matt-bernstein committed:
remove cols from selector

matt-bernstein committed:
cleanup

matt-bernstein committed:
cleanup

matt-bernstein committed:
enable number type

matt-bernstein committed:
fix op precedence

matt-bernstein commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15489321210)
> Successfully merged:  create mode 100644 label_studio/session_policy/migrations/0003_alter_sessiontimeoutpolicy_max_session_age_and_more.py

matt-bernstein committed:
Merge branch 'develop' into 'fb-ROOT-29'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15489321210

matt-bernstein commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15489338819)

matt-bernstein committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15489338819

hlomzik commented on web/libs/datamanager/src/components/Filters/FilterLine/FilterOperation.jsx:
```diff
@@ -56,7 +56,16 @@ export const FilterOperation = observer(({ filter, field, operator, value }) =>
+    // They resolve to the same backend query in this custom case
+    operatorList = operatorList.filter((op) => {
+      const hasEqualOperators = operatorList.some((o) => ["equal", "not_equal"].includes(o.key));
+      const allowedOperators = hasEqualOperators ? ["equal", "not_equal"] : ["contains", "not_contains"];
```
so these vars are the same for any operator, so they can be moved out of this filter as a small improvement

matt-bernstein replied:
Old comment? This doesn't seem true to me, different filters have different operators available, hence the conditional logic

hlomzik commented on web/libs/datamanager/src/stores/Tabs/tab.js:
```diff
@@ -107,6 +116,9 @@ export const Tab = types
     },
 
     get currentFilters() {
+      if (!isFF(FF_ANNOTATION_RESULTS_FILTERING)) {
```
it's not a typo? here you are changing the old behaviour without FF and transforming the old one into new one

matt-bernstein replied:
New columns being added are marked `filter_only`, that option did not previously exist - so with the FF off, new columns will be ignored

hlomzik replied:
ah, that's clever! worth to add a comment it maybe, because it's not obvious

matt-bernstein commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15491599501)
> Successfully merged:  create mode 100644 web/libs/editor/src/tags/object/TimeSeries/README.md

matt-bernstein committed:
Merge branch 'develop' into 'fb-ROOT-29'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15491599501

matt-bernstein commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15491636375)

matt-bernstein committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15491636375

matt-bernstein committed:
Merge branch 'develop' into 'fb-ROOT-29'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15491636375

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-ROOT-29

matt-bernstein committed:
build filter string on fe

matt-bernstein committed:
move internal check to persist control tag columnns on reload

matt-bernstein committed:
rename internal

jombooth committed:
fe code cleanups

jombooth committed:
Merge branch 'fb-ROOT-29' of github.com:HumanSignal/label-studio into fb-ROOT-29

matt-bernstein committed:
don't leak user input across filters

jombooth committed:
relocate filter building responsibility to backend

matt-bernstein committed:
fix filter values leaking across tags due to List type

hlomzik commented on web/libs/datamanager/src/components/Filters/FilterLine/FilterOperation.jsx:
```diff
@@ -56,7 +56,16 @@ export const FilterOperation = observer(({ filter, field, operator, value }) =>
+      const hasEqualOperators = operatorList.some((o) => ["equal", "not_equal"].includes(o.key));
+      const allowedOperators = hasEqualOperators ? ["equal", "not_equal"] : ["contains", "not_contains"];
+      return allowedOperators.includes(op.key);
+    });
```
I meant this:

```suggestion
    const hasEqualOperators = operatorList.some((o) => ["equal", "not_equal"].includes(o.key));
    const allowedOperators = hasEqualOperators ? ["equal", "not_equal"] : ["contains", "not_contains"];
    operatorList = operatorList.filter((op) => allowedOperators.includes(op.key));
```

`allowedOperators` is common for all operators

matt-bernstein committed:
push allowed ops out of filter

Co-authored-by: Andrew <hlomzik@gmail.com>

matt-bernstein committed:
patch label name for control tag filters

matt-bernstein committed:
fix filter search

jombooth committed:
remove filter_only concept in favor of single source of truth

jombooth committed:
exclude self serve

jombooth committed:
fix build

jombooth committed:
unbotch

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-ROOT-29

jombooth commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15621250444)
> Successfully merged:  create mode 100644 web/libs/editor/src/utils/number.js

jombooth committed:
Merge branch 'develop' into 'fb-ROOT-29'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15621250444
